src/new-components/toast/use-toasts.tsx

```diff
@@ -0,0 +1,295 @@
+                setToasts((list) => {
+                    const index = list.findIndex((n) => n.toastId === toastId)
+                    if (index < 0) return list
+                    const copy = [...list]
+                    copy.splice(index, 1)
+                    return copy
+                })
+            })
+        },
+        [animateRemove],
+    )
+
+    const showToast = React.useCallback(
+        function showToast(props: ToastProps) {
+            const toastId = generateElementId('toast')
+            const newToast: InternalToastEntry = {
+                autoDismissDelay: defaultAutoDismissDelay,
+                dismissLabel: defaultDismissLabel,
+                ...props,
+                toastId,
+            }
+            setToasts((list) => [...list, newToast])
+            return () => dismissToast(toastId)
+        },
+        [defaultAutoDismissDelay, defaultDismissLabel, dismissToast],
+    )
+
+    return (
+        <ToastsContext.Provider value={showToast}>
+            {children}
+            <Portal>
+                {toasts.length === 0 ? null : (
+                    <Box
+                        className={styles.stackedToastsView}
+                        position="fixed"
+                        width="full"
+                        paddingX={padding}
+                        paddingBottom={padding}
+                    >
+                        <Stack space="medium">
+                            {toasts.map(({ toastId, ...props }) => (
+                                <InternalToast
+                                    key={toastId}
+                                    ref={mappedRef(toastId)}
+                                    toastId={toastId}
+                                    onDismiss={dismissToast}
+                                    {...props}
+                                />
+                            ))}
+                        </Stack>
+                    </Box>
+                )}
+            </Portal>
+        </ToastsContext.Provider>
+    )
+}
+
+/**
+ * Provides a function `showToast` that shows a new toast every time you call it.
+ *
+ * ```jsx
+ * const showToast = useToasts()
+ *
+ * <button onClick={() => showToast({ message: 'Hello' })}>
+ *   Say hello
+ * </button>
+ * ```
+ *
+ * All toasts fired via this function are rendered in a global fixed location, and stacked one on
+ * top of the other.
+ *
+ * When called, `showToast` returns a function that dismisses the toast when called.
+ *
+ * @see ToastsProvider
+ */
+function useToasts() {
+    return React.useContext(ToastsContext)
+}
+
+/**
+ * Adds a toast to be rendered, stacked alongside any other currently active toasts.
+ *
+ * Internally, this calls `showToast`. It is provided for two reasons:
+ *
+ * 1. Convenience, when you want to fire a toast in markup/jsx code. Keep in mind, though, that
+ *    toasts rendered in this way will be removed from view when ithe context where it is rendered
```

Comment:
```suggestion
 *    toasts rendered in this way will be removed from view when the context where it is rendered
```

Reply:
You've actually made me realize that this sentences is wrong, as it is exactly the opposite. Will update it.